msprime/simulations.py

```diff
@@ -1343,3 +1344,33 @@ def print_history(self, output=sys.stdout):
             print("=" * len(s), file=output)
             self._print_populations(epoch, output)
             print(file=output)
+
+    def popn_size_history(self):
+        """
+        Returns a matrix of population sizes seen
+        by samples from each population at each
+        time defined by the epochs.
+        """
+        num_pops = len(self.epochs[0].populations)
+        pop_size = numpy.zeros((num_pops, len(self.epochs)))
+        popn_set = [[x] for x in range(num_pops)]
+        for j, epoch in enumerate(self.epochs):
+            for k, pop in enumerate(epoch.populations):
+                idx = [k in x for x in popn_set].index(True)
+                pop_size[k, j] = epoch.populations[idx].start_size
+            for event in epoch.demographic_events:
+                if event.type == 'mass_migration' \
+                   event.proportion == 1.0:
+                    print(f"epoch {j} source: {event.source} dest: \
+                          {event.dest}")
+                    popn_set[event.dest] = popn_set[event.dest] + \
+                        popn_set[event.source]
+                    popn_set[event.source] = []
+        return pop_size
+
+    def epoch_times(self):
+        """
+        Helper fxn. Returns appay of epoch times defined by
```

Comment:
The first sentence here is unnecessary (plus, why the abbreviation?). Spelling errorr also: "appay"

Reply:
that's what i get for coding too late at night...